ompi/mca/coll/han/coll_han_alltoallv.c

```diff
@@ -182,9 +251,16 @@ static inline int alltoallv_sendrecv_w_direct_for_debugging(
         if (jloop < nreqs){
             jreq = jloop;
             have_completion = 0;
+            requests[jreq] = MPI_REQUEST_NULL;
         } else {
             have_completion = 1;
-            ompi_request_wait_any( nreqs, requests, &jreq, MPI_STATUS_IGNORE );
+            jreq = jloop%nreqs;
+            if (requests[jreq] == MPI_REQUEST_NULL) {
+                continue;
+            }
+            rc = ompi_request_wait(&requests[jreq], MPI_STATUS_IGNORE);
+            if (rc) break;
```

Comment:
I saw you use this everywhere. Safe programming is good, but can you please add some info about what went wrong instead of just breaking.

Reply:
Added some text hints to help localize the problem.